src/nbl/ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -1139,12 +937,29 @@ SContext::tex_ass_type CMitsubaLoader::cacheTexture(SContext& ctx, uint32_t hier
 					auto contents = viewBundle.getContents();
 					if (!contents.empty())
 						view = core::smart_refctd_ptr_static_cast<asset::ICPUImageView>(contents.begin()[0]);
+					if (view)
+					{
+						auto& image = view->getCreationParameters().image;
+						if (_restore && image->isADummyObjectForCache())
+						{
+							// this will restore the image being kept by found `view`
+							auto bundle = interm_getRestoredAssetInHierarchy(2u, m_assetMgr, tex->bitmap.filename.svalue, ctx.inner.params, hierarchyLevel, ctx.override_);
+							if (bundle.getContents().empty() || image->isADummyObjectForCache())
+							{
+								// if for some reason restore failed, force recreating whole view
+								auto removeBundle = asset::SAssetBundle(nullptr, { view });
+								m_assetMgr->removeAssetFromCache(removeBundle);
+								view = nullptr;
+							}
+						}
+					}
```

Comment:
why isnt this whole thing in some alternative to `IAssetLoaderOverride::findCachedAsset` that does restores from dummy?

Reply:
I understand now.

Reply:
its still very fucked though